pymc3/tests/test_distributions_random.py

```diff
@@ -1306,6 +1305,65 @@ class TestOrderedProbit(BaseTestDistribution):
     ]
 
 
+class TestWishart(BaseTestDistribution):
+    def wishart_rng_fn(self, size, nu, V, rng):
+        return st.wishart.rvs(np.int(nu), V, size=size, random_state=rng)
+
+    pymc_dist = pm.Wishart
+
+    V = np.eye(3)
+    pymc_dist_params = {"nu": 4, "V": V}
+    reference_dist_params = {"nu": 4, "V": V}
+    expected_rv_op_params = {"nu": 4, "V": V}
+    sizes_to_check = [None, (), 1, (1,), 5, (4, 5), (2, 4, 2)]
+    sizes_expected = [
+        (3, 3),
+        (3, 3),
+        (1, 3, 3),
+        (1, 3, 3),
+        (5, 3, 3),
+        (4, 5, 3, 3),
+        (2, 4, 2, 3, 3),
+    ]
```

Comment:
What extra logic (e.g. implemented in `WishartRV`) are all these sizes testing?  

Remember, we don't want the test time to grow larger without proportionally increasing the actual test coverage (and that coverage **doesn't** include NumPy/SciPy's code).  This test module is easily one of the most egregious violators of that principal, so we should always be a little more critical of additions to it.

Reply:
Should that be checking only for edge cases like `None` and `()` ? I just took the `sizes_to_check` list from the original function in `BaseTestDistribution` where the sizes are being checked by default. (And constructed the corresponding expected list.)

Reply:
To echo @kc611, this is not actually testing more than the default test does, just setting the specific parameters of the size test, which usually have to be adapted for multivariate distributions.

Reply:
I agree to both: This implementations is very much in line with the defaults.
The the `BaseTestDistribution` for sure takes a sledgehammer to break a nut, but that problem is out of scope of this PR.

Reply:
Pushing admittedly poor defaults forward is a great way to get into a bad situation, and an even better way to get deeper into existing ones.  

Also, the answer is not one big PR that removes all the testing redundancies; given the knowledge and effort required for that, it's quite unrealistic.  Plus, that won't prevent the problem from reappearing again and again after such a heroic PR.

At the very least, we need to stop adding to the testing redundancies on a *case-by-case basis*, and we can do that by looking for it during reviews; otherwise, we're just hoping that contributors will do that for us, and that's not how things have turned out.


Regarding these exact tests, if they're only targeting `WishartRV.rng_fn`, it looks like we only need a test for an empty and non-empty `size`, because that's the only relevant condition in the code added by this PR.  Every other value besides the first two appears to be testing the code in `RandomVariable` and `scipy.stats.wishart.rvs`.

Reply:
I think there's no harm in not testing the `[ (), (1,), 5,  (2, 4, 2)]` cases if we are testing for `[None, 1, (4, 5)]` cases since they're pretty much the same thing with different values. (Updated accordingly)